x-pack/filebeat/input/httpjson/pagination.go

```diff
@@ -72,7 +72,7 @@ func (p *pagination) nextRequestInfo(ri *requestInfo, response response, lastObj
 
 // getNextLinkFromHeader retrieves the next URL for pagination from the HTTP Header of the response
 func getNextLinkFromHeader(header http.Header, fieldName string, re *regexp.Regexp) (string, error) {
-	links, ok := header[fieldName]
+	links, ok := header[http.CanonicalHeaderKey(fieldName)]
```

Comment:
I think from the docs, `http.Header` keys are always in the canonical form (https://golang.org/pkg/net/http/#Header, https://golang.org/src/net/http/response.go), I added this to take this into account since we are accessing the map directly instead from using the `Get` method which takes care of doing it itself.